assignments/lambda-classes.js

```diff
@@ -1 +1,247 @@
 // CODE here for your Lambda Classes
+class Person{
+    constructor(attributes){
+        this.name = attributes.name
+        this.age = attributes.age
+        this.location = attributes.location
+    }
+    speak(name, location){
+        return `Hello my name is ${this.name}, I am from ${this.location}.`
+    }
+}
+
+class Instructor extends Person{
+    constructor(insAttr){
+        super(insAttr)
+        this.specialty = insAttr.specialty
+        this.favLanguage = insAttr.favLanguage
+        this.catchPhrase = insAttr.catchPhrase
+    }
+    
+    demo(subject){
+        return `Today we are learning about ${subject}.`
+    }
+    
+    grade(Student, subject){
+        return `${Student.name} receives a perfect score on ${subject}`
+    }
+}
+
+class Student extends Person{
+    constructor(stdAttr){
+        super(stdAttr)
+        this.className = stdAttr.className
+        this.favSubjects = stdAttr.favSubjects
+        this.previousBackground = stdAttr.previousBackground
+    }
+
+    listsSubjects(){
+        // return this.favSubjects.forEach({
+        //     return 
+        // })
+            for (let i = 0; i < this.favSubjects.length; i++){
+                console.log(`${this.name} favorite subject is ${this.favSubjects[i]}`)
+            }
+        
+        // return `${this.favSubjects}` 
+    }
+
+    sprintChallenge(subject){
+        return `${Student.name} has begun sprint challenge on ${subject}`
```

Comment:
This works, but better syntax would be: 
```suggestion
        return `${this.name} has begun sprint challenge on ${subject}`
```